src/backend/modules/users/user.service.js

```diff
@@ -1,21 +1,18 @@
 const { UserRepository } = require('./user.repository');
-const db = require('../../database');
 const { DuplicateException } = require('../../common/httpException');
-const { USERS_COLLECTION } = require('../../common/constants/collection.constant');
+const User = require('../../model/user.model');
 
 class Service {
     constructor() {
-        this.UserRepository = UserRepository;
+        this.userRepository = UserRepository;
     }
 
     async createOne(body) {
-        const userRef = db.collection(USERS_COLLECTION);
-
-        const userIsExist = await this.UserRepository.findById(body.username);
-        if (userIsExist) {
-            throw new DuplicateException(`User (${body.username}) is already existed`);
+        const isUserExist = await this.userRepository.findById(body.email);
+        if (isUserExist) {
+            throw new DuplicateException(`User (${body.email}) is already existed`);
         } else {
-            await userRef.doc(body.username).set(body.toJSon());
+            await this.userRepository.createOne(new User(body.email, body.password).toJSon());
```

Comment:
You should convert data from dto to User model and then save it.
1. Create a file dto/createUser.dto.js in user module then create a method to convert data from request body to dto.
example: 
![image](https://user-images.githubusercontent.com/55626329/134323723-8a476377-4de6-4802-892d-2862808d601d.png)
2. In service, convert data from dto to new User object, then pass the user object to createOne method